src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp

```diff
@@ -123,7 +123,12 @@ class boss_sapphiron : public CreatureScript
                 _Reset();
 
                 if (_phase == PHASE_FLIGHT)
+                {
                     ClearIceBlock();
+                    me->SetReactState(REACT_AGGRESSIVE);
+                }
+
+                me->SetFullHealth();
```

Comment:
UpdateVictim() is already used. It's just below `events.Update(diff)` (line 238 in latest).